compliance/unapproved_instance_types/unapproved_instance_types.pt

```diff
@@ -0,0 +1,191 @@
+  label "Email addresses to notify"
+  description "Email addresses of the recipients you wish to notify when new incidents are created"
+end
+
+parameter "param_approved_instance_types" do
+  type "list"
+  label "Approved Instance Types"
+  description "List of approved instance types of AWS, Azure and Google cloud, separated by comma. Example: 'a1.medium,a1.large,a1.xlarge,Standard_A0,Standard_A6,f1-micro,n1-highcpu-2 etc..'"
+end
+
+parameter "param_exclude_tags" do
+  type "list"
+  label "Tags to ignore"
+  description "List of tags that will exclude Servers from being evaluated by this policy. Multiple tags are evaluated as an 'OR' condition. Tag must be of the format 'namespace:predicate=value'. Example: 'rs_agent:type=right_link_lite,rs_monitoring:state=auth'"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "auth_rs", type: "rightscale"
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_instances_list" do
+  request do
+    run_script $js_instances_list, rs_cm_host, rs_project_id, $param_exclude_tags
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "cloud_name", jmes_path(col_item, "links.cloud.name")
+      field "cloud_type", jmes_path(col_item, "links.cloud.cloud_type")
+      field "name", jmes_path(col_item, "name")
+      field "resource_uid", jmes_path(col_item, "resource_uid")
+      field "updated_at", jmes_path(col_item, "timestamps.updated_at")
+      field "instance_type", jmes_path(col_item, "links.instance_type.name")
+      field "href", jmes_path(col_item, "href")
+      field "tags", jmes_path(col_item, "tags")
+      field "state", jmes_path(col_item, "state")
+    end
+  end
+end
+
+datasource "ds_get_unapproved_instances" do
+  run_script $js_get_unapproved_instances, $ds_instances_list, $param_exclude_tags, $param_approved_instance_types
+end
+
+###############################################################################
+# Script
+###############################################################################
+
+script "js_instances_list", type: "javascript" do
+  result "request"
+  parameters "rs_cm_host","rs_project_id","param_exclude_tags"
+  code <<-EOS
+    var filtered_tags = ""
+    if ( param_exclude_tags.length > 0 ){
+      filtered_tags = "&tag!=" + param_exclude_tags.join(",");
+    }
+
+    request = {
+      "auth": "auth_rs",
+      "verb": "GET",
+      "host": rs_cm_host,
+      "path": "/api/instances",
+      "headers": {
+        "X-Api-Version": "1.6",
+        "X-Account": \""+rs_project_id+\"",
+        "accept": "application/json"
+      },
+      "query_params": {
+        "view": "full",
+        "filter": "state=running" + filtered_tags
```

Comment:
the state filter should also look for operational and booting

Reply:
I included operational and booting in the filter. but I can see instances with state operational are getting listed when we mention only 'running' state in the filter.